packages/frontend/src/hooks/useSeason.ts

```diff
@@ -0,0 +1,23 @@
+import { ISeasonDto } from '@etimo/diamonds2-types';
+import { useEffect, useState } from 'react';
+import useFetch from './useFetch';
+
+export const useSeason = (seasonId: string): ISeasonDto => {
+  const {
+    response: fetchedSeason,
+    error,
+    isLoading,
+  } = useFetch(`api/seasons/${seasonId}`, '0');
+  const [season, setSeason] = useState<ISeasonDto>({
+    id: '0',
+    name: '',
+    startDate: new Date(),
+    endDate: new Date(),
+  });
```

Comment:
instead of using a fake value as an initial state you should instead use the generic type `ISeasonDto | null` and set the default value to `null`. This is more consistent and clearly explains the actual state. This way we can easily check for null elsewhere instead of relying on hardcoded default strings.

Reply:
the useSeason-hook might have been redundant, latest commit implements useFetch directly.